File: Dev/CS/Mascaret/Mascaret/VEHA/Kernel/Constraint.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public class Constraint
    {

        private ValueSpecification specification;
        public ValueSpecification Specification
        {
            get { return specification; }
            set { specification = value; }
        }

        Expression expr = null;
        bool result;

        public Constraint()
        {
        }

        public bool eval(Dictionary<string, ValueSpecification> context)
        {
           /* MascaretApplication.Instance.VRComponentFactory.Log("Evaluate constraint : " + context.Count);
            foreach(KeyValuePair<string,ValueSpecification> c in context)
                MascaretApplication.Instance.VRComponentFactory.Log(c.Key);
            * */

            if (expr != null) return result;
            else
            {
                expr = (Expression)(specification);
                ValueSpecification resultVS = expr.evaluateExpression(context);
                result = ((LiteralBoolean)(resultVS)).BValue;
                // MascaretApplication.Instance.VRComponentFactory.Log("Result " + result);
                return result;
            }
        }

    }
}

File: Dev/C++/Mascaret/src/VEHA/Kernel/Constraint.cpp
#include "VEHA/Kernel/Constraint.h"

namespace VEHA
{

Constraint::Constraint()
{
}

Constraint::~Constraint()
{
}
bool Constraint::eval(OclParser::Context context)
{
	if(shared_dynamic_cast<Expression>(getSpecification()))
	{
		return shared_dynamic_cast<Expression>(getSpecification())->evaluateExpression(context);
	}
	else
	{
		bool b=false;
		getSpecification()->getBoolValue(b);
		return b;
	}
	return false;
}

}
